lib/common/ipc_attrd.c

```diff
@@ -103,6 +103,43 @@ send_attrd_request(pcmk_ipc_api_t *api, xmlNode *request)
     return pcmk_rc_ok;
 }
 
+int
+pcmk__attrd_api_clear_failures(pcmk_ipc_api_t *api, const char *host,
+                               const char *resource, const char *operation,
+                               const char *interval_spec, const char *user_name,
+                               int options)
+{
+    int rc = pcmk_rc_ok;
+    xmlNode *request = create_attrd_op(user_name);
+    const char *interval_desc = NULL;
+    const char *op_desc = NULL;
+
+    crm_xml_add(request, PCMK__XA_TASK, PCMK__ATTRD_CMD_CLEAR_FAILURE);
+    crm_xml_add(request, PCMK__XA_ATTR_NODE_NAME, host);
+    crm_xml_add(request, PCMK__XA_ATTR_RESOURCE, resource);
+    crm_xml_add(request, PCMK__XA_ATTR_OPERATION, operation);
+    crm_xml_add(request, PCMK__XA_ATTR_INTERVAL, interval_spec);
+    crm_xml_add_int(request, PCMK__XA_ATTR_IS_REMOTE,
+                    pcmk_is_set(options, pcmk__node_attr_remote));
```

Comment:
You're not passing user_name. I believe all existing callers pass NULL anyway, so we can just drop the argument.

Reply:
pacemaker-attrd will use the actual effective user who initiates the IPC connection if none is given, so end-user tools don't need to set it. The only place it's useful is when a daemon is managing a node attribute on behalf of a user who contacted it (such as crm_resource --cleanup requests going through the controller).

So, we really only need it for API functions that might get used that way, which is probably just the update function.